### PR TITLE
Refractor vcf

### PR DESCRIPTION
**Pull request by moonso**

### CODE REVIEW

**Comment by moonso** [comment]
Hello All! (especially @robinandeer , @guillermo-carrasco , @vasiliosz )
I have reworked alot in the vcf adapter, sorry for the huge PR.
This will add alot of improvements to performance, especially for the SV mode.
Please have a look!!

**Comment by guillermo-carrasco** [comment]
oh my...

File: puzzle/plugins/base_variant_mixin.py [hunk 1]
```diff
@@ -72,17 +73,17 @@ def _add_sv_coordinates(self, variant):
...
-            variant (puzzle.models.variant)
-        """
-        
-        consequences = set()
-        for transcript in variant.transcripts:
-            for consequence in transcript.consequence.split('&'):
-                consequences.add(consequence)
-        
-        variant.consequences = list(consequences)
+    # def _add_consequences(self, variant):
```
Comment (guillermo-carrasco) [inline puzzle/plugins/base_variant_mixin.py]: you can remove the code if you want to, VCS! :-)

Comment (moonso) [inline puzzle/plugins/base_variant_mixin.py]: Haha yes!

Comment (moonso) [inline puzzle/plugins/base_variant_mixin.py]: Fixed

**Comment by vasiliosz** [comment]
Search seems speedy! And opening a variant specific page is much faster. Great work :clap: 

A note is that since gene search is based on string matching, it will return some unexpected results - like searching for _KMT2A_ (aka _MLL_) will also return "associated" genes as listed in the annotation, like: 
(annotated with snpEff)

```
]3:118797604]|sequence_feature|LOW|LPP|ENSG00000145012|site:Breakpoint_for_translocation_to_form_HMGA2-LPP_and_KMT2A/MLL1-LPP|ENST00000543006|protein_coding|9/11|c.1410+49574_1410+49575ins]3:118797604]||||||
```

The important string is
```
site:Breakpoint_for_translocation_to_form_HMGA2-LPP_and_KMT2A/MLL1-LPP
```

This only one of the transcript variants listed in the `ANN` field (of ~10). So I think a gene search for _KMT2A_ in this case will _always_ return all of these associated genes. Perhaps not the expected behaviour of search? Also not sure how to solve it.

**Comment by vasiliosz** [comment]
The listing of genotypes was lost on this branch, but works on master

inklistrad_bild_04_03_16_11_38

**Comment by henrikstranneheim** [comment]
Regarding gene search: One should probably limit the string search to the SNPEFF "|Gene_Name|" or VEP "|SYMBOL|" fields to avoid the associated genes.

**Comment by moonso** [comment]
@vasiliosz , @henrikstranneheim the string match is just a initial filter step.
String matches are superfast in python, in the next step it goes through the potential hits and checks if it is really a hit. (Or this is how it should work :pray: , please note me if id doesen't)

**Comment by moonso** [comment]
@vasiliosz I'll have a look at the genotype thing

**Comment by moonso** [comment]
Btw @vasiliosz or @Bianca-T it would be nice with a test file for SV:s so we can share these kinds of issues

**Comment by vasiliosz** [comment]
@moonso I thought I sent you 1-2? I can share the one I'm working on now privately

**Comment by moonso** [comment]
@vasiliosz yes but we need to have non-sensitive data

**Comment by vasiliosz** [comment]
Well, we could randomly subset them to make them more or less non-sensitive. But this issue wouldn't necessarily be seen. For simple tests, we can use a public WGS trio?

**Comment by moonso** [comment]
Do you know where to find a public one?

**Comment by vasiliosz** [comment]
Maybe 1kg phase 3 data? Could keep as many samples as you want. 
You want it commited, in that case how large? I can prepare it. 

ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502//ALL.chr1.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz

**Comment by moonso** [comment]
Sure, is it SV? We only need like 50-100 variant

**Comment by vasiliosz** [comment]
It’s everything (snvs, indels, svs, others) , but you could easily fish out the ones you want.

**Comment by moonso** [comment]
Ok, sounds perfect @vasiliosz !
I fixed so genotype calls are shown for SV to, thanks for spotting

**Pull request merged**